### PR TITLE
Add preliminary support for pull queues REST API

### PR DESCRIPTION
**Pull request by cdonati**
So far, this just allows inserting, deleting, fetching, and leasing tasks.

### CODE REVIEW

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1784/

File: AppTaskQueue/appscale/taskqueue/distributed_tq.py [hunk 1]
```diff
@@ -128,9 +175,28 @@ def __init__(self):
...
     os.environ['APPLICATION_ID'] = constants.DASHBOARD_APP_ID
 
+    self.db_access = DatastoreProxy()
+
     # Flag to see if code needs to be reloaded.
     self.__force_reload = False
 
+  def get_queue(self, app, queue):
+    """ Fetches a Queue object.
+
```
Comment (menivaitsi) [inline AppTaskQueue/appscale/taskqueue/distributed_tq.py]: Please add Args section.

File: AppTaskQueue/appscale/taskqueue/queue.py [hunk 2]
```diff
@@ -132,6 +139,376 @@ def validate_config(self):
...
+    task.leaseTimestamp = response.lease_expires
+
+    # If ETA was not defined, use the current time.
+    epoch = datetime.datetime.utcfromtimestamp(0)
+    if task.leaseTimestamp == epoch:
+      eta = task.enqueueTimestamp
+    else:
+      eta = task.leaseTimestamp
+
+    # Create an index entry the the task can be queried by ETA. This can't be
```
Comment (menivaitsi) [inline AppTaskQueue/appscale/taskqueue/queue.py]: Change "the the" to something like "so the".

File: AppTaskQueue/appscale/taskqueue/rest_api.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+class RESTTasks(RequestHandler):
+  PATH = '{}/([a-zA-Z0-9-]+)/tasks'.format(REST_PREFIX)
+
+  def initialize(self, queue_handler):
+    """ Provide access to the queue handler. """
+    self.queue_handler = queue_handler
+
+  def get(self, project, queue):
+    """ List all non-deleted task in a queue, whether or not they are
```
Comment (menivaitsi) [inline AppTaskQueue/appscale/taskqueue/rest_api.py]: task -> tasks

File: AppTaskQueue/appscale/taskqueue/task.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+      if not rule(value):
+        raise InvalidTaskInfo(
+          'Invalid task info: {}={}'.format(attribute, value))
+
+  def __repr__(self):
+    """ Generates a string representation of the task.
+
+    Returns:
+      A string representing the task. """
```
Comment (menivaitsi) [inline AppTaskQueue/appscale/taskqueue/task.py]: Please put the closing quotes in a new line.

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1789/

**Pull request merged**